cluster/etcd_service_discovery.go

```diff
@@ -232,16 +239,29 @@ func (sd *etcdServiceDiscovery) GetServersByType(serverType string) (map[string]
 }
 
 func (sd *etcdServiceDiscovery) bootstrap() error {
-	err := sd.bootstrapLease()
-	if err != nil {
+	c := make(chan error)
+	defer close(c)
+	go func() {
+		logger.Log.Infof("waiting for etcd connection")
+		err := sd.bootstrapLease()
+		if err != nil {
+			c <- err
+			return
+		}
+		err = sd.bootstrapServer(sd.server)
+		c <- err
+	}()
+	select {
+	case err := <-c:
```

Comment:
Do you need to signal de application to die gracefully here? Because what happens if this function returns an error is that the startModules function in module.go will raise a fatal error and exit the application

It seems that the behavior on error or timeout should be the same. Given that both cases block the application from really starting.

Reply:
That should be fixed on a next pull request by refactoring the bootstrap/lease logic

Reply:
Ok